### PR TITLE
Java implementation using OkHttp

### PR DESCRIPTION
**Pull request by shashiranjan84**

### CODE REVIEW

**Comment by shashiranjan84** [comment]
one of the four job failed though trvis build worked without any failure in my local

**Comment by shashiranjan84** [comment]
I notice some bug, will send another pull request.

File: src/targets/java/okhttp.js [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+module.exports = function (source, options) {
+  var opts = util._extend({
+    indent: '  '
+  }, options)
+
+  var code = new CodeBuilder(opts.indent)
+
+  var methods = [ 'GET', 'POST', 'PUT', 'DELETE', 'PATCH', 'HEAD']
+
+  var methodsWithBody = [ 'POST', 'PUT', 'DELETE', 'PATCH']
```
Comment (ahmadnassri) [inline src/targets/java/okhttp.js:25]: isn't this redundant? all methods could technically have bodies! should just focus on whether the postData has content instrlead.

Comment (shashiranjan84) [inline src/targets/java/okhttp.js:25]: OkHttp does not support get and head with body plus other methods 'POST',
'PUT', 'DELETE', 'PATCH' need to have body or passed null to api.

http://square.github.io/okhttp/javadoc/index.html

**Pull request merged**